src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -500,6 +510,367 @@ function calc_boundary_flux_by_direction_divergence!(surface_flux_values::Abstra
+    @unpack fstar_upper_threaded, fstar_lower_threaded = cache
+
+    @threaded for mortar in eachmortar(dg, cache)
+        # Choose thread-specific pre-allocated container
+        fstar_upper = fstar_upper_threaded[Threads.threadid()]
+        fstar_lower = fstar_lower_threaded[Threads.threadid()]
+
+        # Calculate fluxes
+        orientation = orientations[mortar]
+        calc_fstar!(fstar_upper, equations_parabolic, surface_flux, u_transformed, dg, u_upper, mortar,
+                    orientation, cache_parabolic)
+        calc_fstar!(fstar_lower, equations_parabolic, surface_flux, u_transformed, dg, u_lower, mortar,
+                    orientation, cache_parabolic)
+
+        mortar_fluxes_to_elements!(surface_flux_values,
+                                   mesh, equations_parabolic, mortar_l2, dg, cache,
+                                   mortar, fstar_upper, fstar_lower)
+    end
+
+    return nothing
+end
+
+@inline function calc_fstar!(destination::AbstractArray{<:Any, 2}, 
+                             equations_parabolic::AbstractEquationsParabolic,
+                             surface_flux, dg::DGSEM,
+                             u_interfaces, interface, orientation,
+                             cache_parabolic)
+    for i in eachnode(dg)
+        # Call pointwise two-point numerical flux function
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u, equations_parabolic, dg, i, interface)
+        #flux = surface_flux(u_ll, u_rr, orientation, equations)
+        # TODO: parabolic; only BR1 at the moment
+        flux = 0.5 * (u_ll + u_rr) # CARE: Not sure if correct!
+
+        # Copy flux to left and right element storage
+        set_node_vars!(destination, flux, equations_parabolic, dg, i)
+    end
+
+    return nothing
+end
+
+@inline function calc_fstar!(destination::AbstractArray{<:Any, 2}, 
+                             equations_parabolic::AbstractEquationsParabolic,
+                             surface_flux, 
+                             u_transformed::AbstractArray,
+                             dg::DGSEM,
+                             u_interfaces, interface, orientation,
+                             cache_parabolic)                      
+    for i in eachnode(dg)
+        # Call pointwise two-point numerical flux function
+        
+        # TODO: Transformed u or regular u? Regular u is used in "calc_gradient!" -> "interface flux"
+        #u_ll, u_rr = get_surface_node_vars(u_transformed, equations_parabolic, dg, i, interface)
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u, equations_parabolic, dg, i, interface)
+
+        # CARE: Not sure if correct!
+        #flux = surface_flux(u_ll, u_rr, orientation, equations_parabolic)
+        # TODO: parabolic; only BR1 at the moment
+        flux = 0.5 * (u_ll + u_rr)
+
+        # Copy flux to left and right element storage
+        set_node_vars!(destination, flux, equations_parabolic, dg, i)
+    end
```

Comment:
Here I am also not sure whether I can use the built-in `set_node_vars!` or whether I need to resort to some manually implemented equivalent as here:

https://github.com/trixi-framework/Trixi.jl/blob/d96514f6d58e48b33816f58e02959d167954fdea/src/solvers/dgsem_tree/dg_2d_parabolic.jl#L569-L573

Reply:
I am not sure. I think I copied and pasted this code from somewhere else, so you may be able to use `set_node_vars!`. 

My suggestion at the moment would be to use this manual implementation until you think it is working, and then see if changing to `set_node_vars!` breaks anything.